iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorAdderImpl.java

```diff
@@ -46,21 +44,106 @@ protected String getTypeDescription() {
         return "Shunt compensator";
     }
 
+    class ShuntCompensatorLinearModelAdderImpl implements ShuntCompensatorLinearModelAdder {
+
+        private double bPerSection = Double.NaN;
+
+        private double gPerSection = Double.NaN;
+
+        private int maximumSectionCount = -1;
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setBPerSection(double bPerSection) {
+            this.bPerSection = bPerSection;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setGPerSection(double gPerSection) {
+            this.gPerSection = gPerSection;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorLinearModelAdder setMaximumSectionCount(int maximumSectionCount) {
+            this.maximumSectionCount = maximumSectionCount;
+            return this;
+        }
+
+        @Override
+        public ShuntCompensatorAdder add() {
+            ValidationUtil.checkLinearBPerSection(ShuntCompensatorAdderImpl.this, bPerSection);
+            ValidationUtil.checkMaximumSectionCount(ShuntCompensatorAdderImpl.this, maximumSectionCount);
+            model = new ShuntCompensatorLinearModelImpl(bPerSection, gPerSection, maximumSectionCount);
+            return ShuntCompensatorAdderImpl.this;
+        }
+    }
+
+    class ShuntCompensatorNonLinearModelAdderImpl implements ShuntCompensatorNonLinearModelAdder {
+
+        private final List<ShuntCompensatorNonLinearModelImpl.SectionImpl> sections = new ArrayList<>();
+        private boolean hasDefinedG = false;
```

Comment:
Is it just for debugging?

Reply:
No, this is a bit complicated for what it does, but it is to check that if the conductance of one section has been defined, all sections must have defined conductance (and vice versa, if a section with an undefined conductance has been added, all sections must have undefined conductances).